assets/js/googlesitekit/modules/datastore/settings-panel.test.js

```diff
@@ -0,0 +1,156 @@
+	} );
+
+	describe( 'reducer', () => {
+		it( 'has the appropriate initial state', () => {
+			expect( Object.keys( store.getState() ) ).toContain( 'settingsPanel' );
+			expect( store.getState().settingsPanel ).toEqual( expectedInitialState );
+		} );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setModuleSettingsPanelState', () => {
+			it( 'sets module settings panel state', () => {
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'view' );
+				expect( store.getState().settingsPanel ).toEqual( { ...expectedInitialState, currentModule: slug, isEditing: false } );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'edit' );
+				expect( store.getState().settingsPanel ).toEqual( { ...expectedInitialState, currentModule: slug, isEditing: true } );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'closed' );
+				expect( store.getState().settingsPanel ).toEqual( { ...expectedInitialState, currentModule: null, isEditing: false } );
+			} );
+
+			it( 'should not change the panel state when called with an invalid value', () => {
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'edit' );
+				const expectedState = { ... store.getState().settingsPanel };
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'invalid' );
+				} ).toThrow( 'value should be one of closed,edit,view' );
+				expect( store.getState().settingsPanel ).toEqual( expectedState );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getModuleSettingsPanelState', () => {
+			it( 'returns "closed" by default', () => {
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'closed' );
+			} );
+
+			it( 'returns the settings panel state for a module', () => {
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'view' );
+				registry.select( STORE_NAME ).getModuleSettingsPanelState( slug );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'view' );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'closed' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'closed' );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'edit' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'edit' );
+			} );
+		} );
+
+		describe( 'isModuleSettingsPanelOpen', () => {
+			it( 'returns true when module settings panel is in view state for a given module', () => {
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'view' );
+				expect( registry.select( STORE_NAME ).isModuleSettingsPanelOpen( slug ) ).toBe( true );
+			} );
+
+			it( 'returns true when module settings panel is in edit state for a given module', () => {
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'view' );
```

Comment:
Oops, this test is using `view` as well instead of `edit`.
```suggestion
				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'edit' );
```